### PR TITLE
Polish

### PR DESCRIPTION
**Pull request by OLPMO**
Simplify the code and improve execution efficiency of the code with Java8

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/AutoConfigurationSorterTests.java [hunk 1]
```diff
@@ -17,11 +17,7 @@
 package org.springframework.boot.autoconfigure;
 
 import java.io.IOException;
-import java.util.Arrays;
-import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Properties;
-import java.util.Set;
+import java.util.*;
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/AutoConfigurationSorterTests.java]: We don't use wildcard imports. Please revert this change

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceInitializer.java [hunk 2]
```diff
@@ -125,10 +125,7 @@ private boolean isEnabled() {
 		if (mode == DataSourceInitializationMode.NEVER) {
 			return false;
 		}
-		if (mode == DataSourceInitializationMode.EMBEDDED && !isEmbedded()) {
-			return false;
-		}
-		return true;
+		return mode != DataSourceInitializationMode.EMBEDDED || isEmbedded();
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceInitializer.java]: When there are multiple if statements like this, I find the current version more readable than collapsing the last if statement as you suggest.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/AutoConfigurationSorter.java [hunk 3]
```diff
@@ -136,8 +136,7 @@ AutoConfigurationClass get(String className) {
 		}
 
 		Set<String> getClassesRequestedAfter(String className) {
-			Set<String> classesRequestedAfter = new LinkedHashSet<>();
-			classesRequestedAfter.addAll(get(className).getAfter());
+			Set<String> classesRequestedAfter = new LinkedHashSet<>(get(className).getAfter());
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/AutoConfigurationSorter.java]: I find personally the previous version more readable: nesting the method call in the constructor parameter makes it harder to read.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ImportAutoConfigurationImportSelector.java [hunk 4]
```diff
@@ -58,8 +58,7 @@
 
 	@Override
 	public Set<Object> determineImports(AnnotationMetadata metadata) {
-		Set<String> result = new LinkedHashSet<>();
-		result.addAll(getCandidateConfigurations(metadata, null));
+		Set<String> result = new LinkedHashSet<>(getCandidateConfigurations(metadata, null));
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ImportAutoConfigurationImportSelector.java]: I find personally the previous version more readable: nesting the method call in the constructor parameter makes it harder to read.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jmx/ParentAwareNamingStrategy.java [hunk 5]
```diff
@@ -57,8 +57,7 @@ public void setEnsureUniqueRuntimeObjectNames(boolean ensureUniqueRuntimeObjectN
 	@Override
 	public ObjectName getObjectName(Object managedBean, String beanKey) throws MalformedObjectNameException {
 		ObjectName name = super.getObjectName(managedBean, beanKey);
-		Hashtable<String, String> properties = new Hashtable<>();
-		properties.putAll(name.getKeyPropertyList());
+		Hashtable<String, String> properties = new Hashtable<>(name.getKeyPropertyList());
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jmx/ParentAwareNamingStrategy.java]: I find personally the previous version more readable: nesting the method call in the constructor parameter makes it harder to read.

Comment (OLPMO) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jmx/ParentAwareNamingStrategy.java]: I think the previous version of the code will cause the resize of the hash table, so i commit these change.Is my thought something wrong?

File: spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/AutoConfigurationSorterTests.java (hunk 1 above)
Comment (OLPMO) [reply spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/AutoConfigurationSorterTests.java]: oh,sorry!It is my carelessness.I will correct it immediately.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ImportAutoConfigurationImportSelector.java (hunk 4 above)
Comment (OLPMO) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ImportAutoConfigurationImportSelector.java]: Create the hashset with default size ,and then put all the item in it.Would not it cause the hash table resize?

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/AutoConfigurationSorter.java (hunk 3 above)
Comment (OLPMO) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/AutoConfigurationSorter.java]: Create the hashset with default size ,and then put all the item in it.Would not it cause the hash table resize?

**Comment by OLPMO** [comment]
Is there anything else I need to improve on in my pr? @snicoll

**Comment by snicoll** [comment]
@OLPMO thanks for the follow-up. I did asked for feedback and @mbhave shared an interesting idea. We could keep your change with the same readability by introducing a variable prior to creating the collection. Can you please update your PR in that direction?

**Comment by OLPMO** [comment]
@snicoll I have submitted my code in order to increase the readability of the code.It seems like that my submitted code did not pass ci.However, my local code is able to build successfully.Is there anything wrong with my code?

**Comment by snicoll** [comment]
@OLPMO thank you for making your first contribution to Spring Boot and following-up so quickly on the feedback to your proposal.

To answer your previous question, nothing wrong with the code, the CI failure is unrelated. Thanks again.

**Comment by OLPMO** [comment]
@snicoll  You‘re welcome
